fn/fn.go

```diff
@@ -0,0 +1,53 @@
+package fn
+
+// Unit is a type alias for the empty struct to make it a bit less noisy to
+// communicate the informationaless type.
+type Unit = struct{}
+
+// Comp is left to right function composition. Comp(f, g)(x) == g(f(x)). This
+// can make it easier to create on the fly closures that we may use as
+// arguments to other functions defined in this package (or otherwise).
+func Comp[A, B, C any](f func(A) B, g func(B) C) func(A) C {
+	return func(a A) C {
+		return g(f(a))
+	}
+}
+
+// Iden is the left and right identity of Comp. It is a function that simply
+// returns its argument. The utility of this function is only apparent in
+// conjunction with other functions in this package.
+func Iden[A any](a A) A {
+	return a
+}
+
+// Curry takes a two argument function and returns a function that accepts
+// the first argument and then returns a function that accepts the second
+// argument. This can be a useful utility when taking functions defined in a
+// typical go style and adapting them to work with higher-order functions that
+// expect functions of a single argument.
+func Curry[A, B, C any](f func(A, B) C) func(A) func(B) C {
```

Comment:
Could be cut perhaps? Main thing that jumps to mind is syntax bloat, same with `Uncurry`.